### ISSUE TITLE
Condition rule is not working if the field which should match belongs to another page.

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#521]
**Description**
In a multi-pages form, the condition rule is not working if the field which should match belongs to another page.
This happens when the Page reload (server-side) option is enabled. If the matching field is within the same page, it works.

**Steps to reproduce**
1. Setup a multi-pages form
2. On the first page, add any field
3. On the second page, add any field
4. On this field, add a rule to show the field if its value matches the field on the first page

### Form settings
- Multi-page form: Yes
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: dev-craft-3 as 1.4.9
- Craft version: 3.6.17
- Multi-site: No

**Comment by engram-design** [comment]
Yep, this is a great point for a Page Reload-based form in that each page isn't loaded, only the current page - unlike Ajax-based forms. As such, when it tries to grab the value from the field from a previous step, it doesn't exist.

I'll need to give this some further thought.

**Comment by romainpoirier** [comment]
I'm sorry to ask you this, but do you have any information about when you plan to fix this? Thank you!

**Comment by engram-design** [comment]
It's not a quick fix sorry. I need to come up with a solution that shows previous pages' field content, alongside the current page, like an Ajax form does. I just need to carefully test the implications of this before rolling out this change in behaviour.

**Comment by romainpoirier** [comment]
Ok thank you, I understand.

Do you think that there's currently a workaround using `craft.formie.renderForm` rendering options?
Like manually set _Field Options_ (`formDataAttributes`) based on the conditions of the field of reference?

Simple case: an hidden _Agree_ field on page 1, on page 2, load the current submission and populate field condition (display / hide) based on the field on page 1 using the _Field Options_ on page 2's field.

**Comment by engram-design** [comment]
The only real workaround at the moment is to use Ajax-submit I'm afraid. The conditions JS is heavily tied to the DOM, so it'll look for the inputs via their `name` attribute (so `<input name="fields[sourceField]">`. As they don't exist in a Page Reload form, it doesn't calculate this.

**Comment by romainpoirier** [comment]
Right now, we can still setup conditions on back-end, but it's still not working on front-end using the page reload setup.
There is also no warning about this in the documentation: https://verbb.io/craft-plugins/formie/docs/feature-tour/conditions. Maybe you should temporary update this.

This is a serious issue for us, as we can't use the AJAX submission method, and we really require the conditions feature on fields + pages. Before we started working on, we thought we could.

I'm really sorry to ask you again, but can't you get any approximate deadline for the resolution of this bug? Do you think it would be days, weeks, or months...? We have a strict deadline for the project we're working on and have to be very careful about that bugged feature which is imperative for our complex form.

Once again, thank you very much for the help provided.

**Comment by engram-design** [comment]
I would expect to be able to address this within a few weeks. I've already been testing out a few methods to achieve this. I can't stress enough this isn't a quick fix and there might even be breaking changes, so I'm testing a few options. I'll be sure to let you know when there's something available to test very soon.

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.12",
  "...": "..."
}
```

Then run `composer update`.

This change in behaviour now renders all fields and pages on the form, rather than the previous behaviour of just rendering the current page. Ajax-based forms always rendered the entire form, just toggling hiding each page, so now the two behaviours are the same. The actual change is simple, but I wanted ample time to be able to properly test this as it's sort of a breaking change. At least a change in behaviour. Thanks for your patience - let me know if that doesn't work for you.

**Comment by romainpoirier** [comment]
As far as I've tested right now is it working fine. Thank you so much for answered quickly to this needed one!

I've just encountered probably a non-related issue, but sometimes on the current page where the conditional fields belongs, the condition is not checked on tabs when these fields are edited several times using JS. This happens in a very specific case where I'm checking some checkboxes fields using an external script (using `$(element).prop('checked', false)` or `$(element).prop('checked', true)`).

**Comment by engram-design** [comment]
Hmm, sounds like a new issue. Interesting though, I'll have to run some tests. You're using jQuery?

**Comment by romainpoirier** [comment]
Yes of course, for `$(element).prop('checked')` I'm using jQuery (here using `3.6.0` version). Thank you!

**Comment by engram-design** [comment]
Just looking at it now, and we check for the `click` event for checkbox and radio inputs - https://github.com/verbb/formie/blob/6b3a9fda925e7d7eb93d714ba02e6b8761f2a8ed/src/web/assets/frontend/src/js/fields/conditions.js#L186-L203

So you might need to trigger a `.trigger('click')` after settings those.

I believe I had some edge-case issues with detecting "change" on those inputs.

**Comment by romainpoirier** [comment]
Thank you, I can confirm that it works using `.click()` instead of `.prop('checked')`.

**Comment by engram-design** [comment]
Fixed in [1.4.13](https://github.com/verbb/formie/releases/tag/1.4.13)
